doc/memory_tune.md

```diff
@@ -30,3 +30,4 @@ Normally there are some methods to tune the memory usage:
 - set the app heap size with `wasm_runtime_instantiate`
 - use `nostdlib` mode, add `-Wl,--strip-all`: refer to [How to reduce the footprint](./build_wasm_app.md#2-how-to-reduce-the-footprint) of building wasm app for more details
 - use XIP mode, refer to [WAMR XIP (Execution In Place) feature introduction](./xip.md) for more details
+- when using the Wasm C API, set `clone_wasm_binary` in `LoadArgs` to `false` and free the wasm binary buffer (with `wasm_byte_vec_delete`) before wasm module execution 
```

Comment:
May be invalid? As the wasm/aot file may be referred by wasm module after loading, like re-using strings in wasm file:

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_runtime.c#L4596-L4604

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/aot/aot_loader.c#L324-L330

Reply:
It shouldn't be a problem if the module buffer is freed just before wasm execution (that is, `wasm_func_call`), not immediately after loading. Otherwise other APIs used after loading e.g. https://github.com/bytecodealliance/wasm-micro-runtime/blob/ca61184ced7d885395bc2425c87f0d063935dd1e/core/iwasm/include/wasm_c_api.h#L546
will try to use it and crash:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/ca61184ced7d885395bc2425c87f0d063935dd1e/core/iwasm/common/wasm_c_api.c#L2722

Reply:
Not very sure, it is a little confusing that execution won't use some fields of module? e.g. dumping call stacks when exception is thrown may get the function name from `func_inst->u.func_import->field_name`, `func_inst->u.func->field_name` or `export_func->name`:

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_runtime.c#L3265-L3269

Reply:
True, I didn't consider that case. Would it be enough to pass `is_load_from_file_buf` as false to cover that case?

Reply:
>Would it be enough to pass is_load_from_file_buf as false to cover that case?

Tried it now (passing `false` as `is_load_from_file_buf` to `load_from_sections`) and it seems to work for the call stack when an exception is thrown. Is there any other case that I'm not considering?

Reply:
Seems yes. I am also thinking that we should add a flag in LoadArgs to control the is_load_from_file_buf flag inside wasm_loader.c and aot_loader.c, e.g. add an extra argument for wasm_load, wasm_loader_load and aot_loader_from_aot_file, allowing them to pass `is_load_from_file_buf = false` to wasm_const_str_list_insert.

And then change the behavior of wasm_runtime_load_ex accordingly. And for wasm_runtime_load, set the right extra flag in LoadArgs before calling wasm_runtime_load_ex to allow modifying to the buffer, or, keep the same behavior for wasm_runtime_load before modification.

Reply:
>I am also thinking that we should add a flag in LoadArgs to control the is_load_from_file_buf flag

Can't we use `clone_wasm_binary` for that? Or do you think we will need the distinction for other use cases in the future?

Reply:
Also because `clone_wasm_binary=false` wouldn't work properly without `is_load_from_file_buf=false`

Reply:
It is a different concept from clone_wasm_binary: wasm loader doesn't keep another binary now no matter clone_wasm_binary is true or false, it just modifies the binary and refers to it after loading. Using the flag for wasm loader makes confusing.

How about adding `bool wasm_binary_freeable` in LoadArgs, and mentioning that it is only for wasm/aot loader? And:
- Change wasm_runtime_load_ex, aot_load_from_aot_file, wasm_loader_load, wasm_load to use the flag
- Change wasm_runtime_load and wasm_module_new_ex, pass correct `wasm_binary_freeable` flag to wasm_runtime_load_ex

Or do you have better name for the flag?

Reply:
I think I applied your suggestions, let me know if I missed or misunderstood anything

Reply:
OK, I added some comments, could you help check and resolve them? I will check it again then.

Reply:
Thanks for the review, I pushed the suggested changes

Reply:
Yes, it seems that all the places have been updated. It LGTM except a comment to update the document. Let's wait for comments from other developers before merging it.

Reply:
BTW, had better try more cases. And it seems that we can close #3377 after this PR is merged?

Reply:
>And it seems that we can close https://github.com/bytecodealliance/wasm-micro-runtime/pull/3377 after this PR is merged?

Yes, this PR does what we were looking for in the initial issue https://github.com/bytecodealliance/wasm-micro-runtime/issues/3354